modules/tfm/tfm/common/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if(TFM_PARTITION_PLATFORM)
+  set(src ${CMAKE_CURRENT_LIST_DIR}/src)
+  set(nrf_dir ${CMAKE_CURRENT_LIST_DIR}/../../../../)
+
+  target_include_directories(platform_s
+    PUBLIC
+    include
+    ${nrf_dir}/include/tfm
+    )
+
+  target_sources(platform_s
+    PRIVATE
+    ${src}/tfm_platform_system.c
+    ${src}/tfm_ioctl_s_api.c
+    )
```

Comment:
This correctly adds the files you want to the library, but originally you also did:
https://github.com/nrfconnect/sdk-nrf/blob/6df8a340532a4fc2cf78fe22ca51bed750781623/modules/tfm/common/CMakeLists.txt#L12-L14

so I assume you intend to avoid getting the file `tfm_platform_system.c` added to the lib here:
```
        $<$<BOOL:${TFM_PARTITION_PLATFORM}>:${CMAKE_CURRENT_SOURCE_DIR}/services/src/tfm_platform_system.c>
```
[nordic_nrf/common/nrf9160/CMakeLists.txt#L47](https://github.com/hakonfam/sdk-trusted-firmware-m/blob/cfdd2eb5633327b78c8c5f52fa74ad09780449e9/trusted-firmware-m/platform/ext/target/nordic_nrf/common/nrf9160/CMakeLists.txt#L47) in the first place.

So I assume you are missing a:
```
set(TFM_PARTITION_PLATFORM False)

add_subdirectory(${Trusted\ Firmware\ M_SOURCE_DIR}/platform/ext/target/nordic_nrf/common/nrf9160 nrf9160)
```
in this file: https://github.com/hakonfam/fw-nrfconnect-nrf-1/blob/secure_read/modules/tfm/tfm/boards/nrf9160/CMakeLists.txt

Also, it appears to me that `tfm/common/CMakeLists.txt` no longer gets included in the build.
is that correctly observer ?

Reply:
I mistakenly pushed while in the progress of trying out a solution without duplication, sorry for the inconvenience. Should be fixed now.